bindist/wrapper.in

```diff
@@ -73,7 +77,7 @@ check_ghc() {
 			return 1
 		fi
 		PKGCONF="${check_ghc_libdir}/package.conf.d"
-		MY_ABI_HASHES="$(for dep in $("${GHC_PKG}" --global --global-package-db "$PKGCONF" list --simple-output) ; do printf "%s:" "${dep}" && "${GHC_PKG}" --global --global-package-db "$PKGCONF" field "${dep}" abi --simple-output ; done | tr '\n' ' ' | xargs)"
+		MY_ABI_HASHES="$(for dep in ${BOOT_PKGS} ; do printf "%s:" "${dep}" && "${GHC_PKG}" --global --global-package-db "$PKGCONF" field "${dep}" abi --simple-output ; done | tr '\n' ' ' | xargs)"
```

Comment:
Where exactly do you allow a subset of packages here? You check everything that's used at compile time. Even if it's not relevant to HLS and if it's not an actual boot package. That's not a subset. It's either exactly all boot libraries or a superset of them.

I think a better idea is really to:

1. hardcode the pkgs that are relevant to HLS ABI
2. Only check those

I'd also be fine to hardcode the entire list of boot libraries. These may diverge for GHC versions in the future, but we only support a certain range anyway.

Reply:
I am so, so, sorry. I had a stupid thinko/sign-error in the commit title. I meant all along, that the runtime packages can be a superset, not a subset of the compiletime/bootpkgs.

I think using the BOOT_PKGS variable in the wrapper script is the right thing and matching on all boot pkgs is also imo the right thing, because probably most packages are relevant to the HLS or TH ABI and figuring out which exactly aren’t could be annoying and probably not worth it. (At least that’s what Zubin argued for.)

Then we can argue about how to populate it. I think hardcoding it is simply more work, especially if we hardcode the versions, too. In all the cases (there are two) that I know where the wrapper is populated, the compiletime ghc-pkgs are exactly the boot pkgs, so I wouldn‘t worry about using that to populate the BOOT_PKGS variable …

Reply:
So, what I prefer is:

1. hardcode boot pkgs in bootstrap script
2. during runtime throw away all non boot packages when comparing ABI

Reply:
I think I misunderstood the approach a little. I thought we still use the whole global DB during runtime.

Anyway... one could argue this still allows the ABI check to over-match, but building the bindists via the Makefile is not supposed to be done by the user. So it's up to the distributor to not mess it up.

As such, I think this is fine.